lib/peopleHR.ts

```diff
@@ -0,0 +1,235 @@
+
+    document.body.innerHTML = text;
+
+    blocklist.forEach((tag) => {
+        const elements = document.querySelectorAll(tag);
+        elements.forEach((element) => {
+            element.remove();
+        });
+    });
+
+    return document.body.innerHTML;
+}
+
+/**
+ * Formats HTML with a virtual DOM and Regex.
+ * @param description A string of HTML markup
+ * @returns A formatted HTML string
+ */
+export function cleanUpHTML(description: string): string {
+    const blocklist = ['script', 'style', 'iframe', 'br'];
+    //We can't remove the font tags this way as it will delete their child paragraphs.
+    description = removeAllElementsInBlocklist(description, blocklist);
+
+    //Remove any zero width spaces (causes a blank new line in some <p> tags)
+    description = description.replace(/\u200B/g, '');
+
+    //Replace strong emphasis tags with bold italic text.
+    description = description.replace(
+        /<strong><[^\<]+<em>([^\<]+)<\/em><\/font><\/strong>/gm,
+        `<b class="bold-italic-rich-text"><i>$1</i></b>`,
+    );
+
+    //Replace bold italic styling with bold italic text.
+    description = description.replace(
+        /<[^\<]+font-weight: 700; font-style: italic;[^\>]*>([^\<]+)\<\/font>/gm,
+        `<b class="bold-italic-rich-text"><i>$1</i></b>`,
+    );
+
+    //Replace bold font decelerations with bold text.
+    description = description.replace(
+        /\<[^\<]+font-weight: 700;[^\>]+\>([^\<]+)<\/font>/gm,
+        `<b class="bold-rich-text">$1</b>`,
+    );
+
+    //Replace any italic font decelerations with italic text.
+    description = description.replace(
+        /\<[^\<]+font-style: italic;[^\>]+\>([^\<]+)<\/font>/gm,
+        `<i class="italic-rich-text">$1</i>`,
+    );
+
+    //Remove all font tags and any styling of text.
+    description = description.replace(/style="[^"]+"/gm, '');
+    description = description.replace(/\<font[^\>]+\>/gm, '');
+    description = description.replace(/\<\/font\>/gm, '');
+
+    //Remove role=presentation and aria-level=1 from list items
+    description = description.replace(/\s*role="presentation"\s*/gm, '');
+    description = description.replace(/\s*aria-level="1"\s*/gm, '');
+
+    //Remove excess attributes from paragraph tags
+    description = description.replace(/\s*dir="ltr"\s*/gm, '');
+    description = description.replace(/\s*_rdeditor_exists="1"\s*/gm, '');
+
+    //Remove any empty paragraph tags (these shouldn't cause whitespace anyway, but just in case)
+    description = description.replace(/\<p[ ]*\>[ ]*\<\/p\>/gm, '');
+    //In the first job posting there are two levels of empty tags to remove.
+    description = description.replace(/\<p[ ]*\>[ ]*\<\/p\>/gm, '');
+
+    return description;
+}
+
+function invalidValue(value: any) {
+    if (value === '') return true;
+    if (!value) return true;
+    return false;
+}
+
+function jobPostingJSONIsValid(json: any) {
+    //To be refactored with neater code and Sentry warnings
```

Comment:
suggestion: could be worth using a JSON schema validator or something like https://github.com/jquense/yup instead of writing this custom?